src/components/SubgraphExplorer.tsx

```diff
@@ -155,12 +164,75 @@ query MyQuery {
     }
   };
 
-  const updateQuery = (query: string | undefined) => {
-    setQuery(query ?? "");
+  const updateQuery = (query: string | undefined) => setQuery(query ?? "");
+
+  const graphiQLFetcher = async (graphQLParams: FetcherParams) => {
+    const parsedParams = parseGraphQLParams(graphQLParams);
+
+    const data = await fetch(subgraphUrl, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(parsedParams),
+      credentials: "same-origin",
+    });
+    setFetchedFromUrl(subgraphUrl);
+    return data.json().catch(() => data.text());
+  };
+
+  const parseGraphQLParams = (graphQLParams: FetcherParams): FetcherParams => {
+    const { query } = graphQLParams;
+
+    const parsedQuery = lowercaseAddress(query);
+    const parsedVariables = lowercaseAddress(variables || "");
+
+    if (parsedQuery !== query) setQuery(parsedQuery);
+    if (parsedVariables !== variables) setVariables(parsedVariables);
+
+    if (parsedQuery !== query || parsedVariables !== variables) {
+      setShowAddressToast(true);
+    }
+
+    return {
+      ...graphQLParams,
+      query: parsedQuery,
+      variables: JSON.parse(parsedVariables),
+    };
+  };
+
+  const lowercaseAddress = (str: string) =>
+    str.replace(ADDRESS_REGEX, (v, _p1, p2) => v.replace(p2, p2.toLowerCase()));
+
+  const handleCloseToast = (
+    _e: SyntheticEvent | Event,
+    reason?: SnackbarCloseReason
+  ) => {
+    if (reason === "clickaway") return;
+    setShowAddressToast(false);
   };
 
   return (
     <>
+      <Snackbar
+        open={showAddressToast}
+        autoHideDuration={8000}
+        onClose={handleCloseToast}
+        message="Automatically lower cased addresses in your query and variables"
```

Comment:
Nitpick but instead of "lower cased" I would say "converted to lower case". I'm not sure "lower cased" is actually used as a verb.

If the sentence becomes too long then you could just say: "Addresses automatically converted to lower case"

Reply:
Left it as "Addresses in your query and variables automatically converted to lower case" atm.